flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/dataset/ClusterDataSetPartitionListHandler.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.handler.dataset;
+
+import org.apache.flink.api.common.time.Time;
+import org.apache.flink.runtime.resourcemanager.ResourceManagerGateway;
+import org.apache.flink.runtime.rest.handler.HandlerRequest;
+import org.apache.flink.runtime.rest.handler.RestHandlerException;
+import org.apache.flink.runtime.rest.handler.resourcemanager.AbstractResourceManagerHandler;
+import org.apache.flink.runtime.rest.messages.EmptyMessageParameters;
+import org.apache.flink.runtime.rest.messages.EmptyRequestBody;
+import org.apache.flink.runtime.rest.messages.dataset.ClusterDataSetListHeaders;
+import org.apache.flink.runtime.rest.messages.dataset.ClusterDataSetListResponseBody;
+import org.apache.flink.runtime.webmonitor.RestfulGateway;
+import org.apache.flink.runtime.webmonitor.retriever.GatewayRetriever;
+
+import javax.annotation.Nonnull;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Handler for {@link ClusterDataSetListHeaders}.
+ */
+public class ClusterDataSetPartitionListHandler extends AbstractResourceManagerHandler<RestfulGateway, EmptyRequestBody, ClusterDataSetListResponseBody, EmptyMessageParameters> {
```

Comment:
It seems we are listing data sets and not partitions. Maybe `ClusterDataSetListHandler` would be a better name. 

I also wanted to ask if we are no longer implementing the Public Interface as described in the FLIP: https://cwiki.apache.org/confluence/display/FLINK/FLIP-67%3A+Cluster+partitions+lifecycle

Reply:
I have adjusted the name, this was an oversight.

There are indeed some differences to the proposed API in the FLIP, let's go through them:

1) lack of some meta data like job name, creation dates and such.

These may be added later. We made sure that existing code can be extended easily to carry additional meta-data to the ResourceManager, but left it out for simplicity's sake. We only retained data that we needed functionality-wise.

2) rename of URLs to use `dataset` instead of `partition`

This was an oversight in terminology that was noticed way too late. Back then we used the term "cluster partition" to both denote a partition of a data set and the data set as a whole. This is naturally confusing, we I introduced a new term. "Cluster dataset" was the logical choice.

3) Lack of GET/dataset/:datasetid

This was left out because it is not functionally required for how we envision `Interactive Programming` to work (because it wouldn't retrieve partition descriptors through the REST API).
Additionally, the partition descriptors may be too large/numerous to safely transmit them over heartbeats; meaning we may need a separate transfer mechanism.
It is still something that we should _eventually_ have imo so we don't limit usage of cluster data sets to `Interactive Programming`.

Reply:
ok